python/pyarrow/tests/parquet/test_dataset.py

```diff
@@ -972,6 +972,10 @@ def test_dataset_read_pandas(tempdir, use_legacy_dataset):
 
     tm.assert_frame_equal(result, expected)
 
+    # also be able to pass the columns as a set (ARROW-12314)
+    result = dataset.read_pandas(columns=set(columns)).to_pandas()
+    tm.assert_frame_equal(result, expected)
```

Comment:
It looks like this test needs to account for column order potentially being different, since sets (surprisingly?) didn't get deterministic ordering like dictionaries did.

Reply:
Yes, still needed to fix that, will do that now